core/domain/story_jobs_one_off.py

```diff
@@ -150,3 +150,45 @@ def reduce(key, values):
                 sum(ast.literal_eval(v) for v in values))])
         else:
             yield (key, values)
+
+
+class StoryExplorationsAuditOneOffJob(jobs.BaseMapReduceOneOffJobManager):
+    """One-off job to validate explorations in stories."""
+
+    _DELETED_KEY = 'story_deleted'
+    _PROCESSED_KEY = 'story_processed'
+    _ERROR_KEY = 'story_failed_validation'
+
+    @classmethod
+    def entity_classes_to_map_over(cls):
+        return [story_models.StoryModel]
+
+    @staticmethod
+    def map(item):
+        if item.deleted:
+            yield (StoryExplorationsAuditOneOffJob._DELETED_KEY, 1)
+            return
+
+        story = story_fetchers.get_story_by_id(item.id)
+        exp_ids = story.story_contents.get_all_linked_exp_ids()
+        validation_errors = story_services.validate_explorations_for_story(
+            exp_ids, False)
+        if validation_errors:
+            yield (
+                StoryExplorationsAuditOneOffJob._ERROR_KEY,
+                'Failed to validate explorations for story %s: %s' % (
+                    item.id, validation_errors))
+            return
+
+        yield (StoryExplorationsAuditOneOffJob._PROCESSED_KEY, 1)
+
+    @staticmethod
+    def reduce(key, values):
+        if key == StoryExplorationsAuditOneOffJob._DELETED_KEY:
+            yield (key, ['Encountered %d deleted stories.' % (
+                sum(ast.literal_eval(v) for v in values))])
+        elif key == StoryExplorationsAuditOneOffJob._PROCESSED_KEY:
+            yield (key, ['Successfully processed %d stories.' % (
+                sum(ast.literal_eval(v) for v in values))])
+        else:
+            yield (key, values)
```

Comment:
I think this is a little more readable:
```suggestion
        if key == StoryExplorationsAuditOneOffJob._DELETED_KEY:
            num_deleted = sum(ast.literal_eval(v) for v in values)
            values = ['Encountered %d deleted stories.' % num_deleted]
        elif key == StoryExplorationsAuditOneOffJob._PROCESSED_KEY:
            num_processed = sum(ast.literal_eval(v) for v in values)
            values = ['Successfully processed %d stories.' % num_processed]
        yield (key, values)
```

Reply:
Done